psi4/driver/p4util/spectrum.py

```diff
@@ -0,0 +1,376 @@
+
+    Parameters
+    ----------
+    domain : Union[List[float], np.ndarray]
+        The domain of the Gaussian profile.
+    gamma : float
+        Broadening parameter.
+        This is related to the full width at half maximum as :math:`\mathrm{FWHM} = \gamma \sqrt{2\ln 2}`
+
+    Notes
+    -----
+    Use this profile to model inhomegenous broadening.
+    """
+
+    def lineshape(self, x_0: float) -> np.ndarray:
+        """Gaussian function on `self.domain`, centered at `x_0` with broadening `self.gamma`.
+
+        Parameters
+        ----------
+        x_0 : float
+            Center of the Gaussian, i.e. its maximum.
+
+        Returns
+        -------
+        gauss : np.ndarray
+            The Gaussian profile.
+        """
+        prefactor = 2.0 / (self.gamma(x_0) * np.sqrt(2.0 * np.pi))
+        exponent = -2.0 * ((self.domain - x_0) / self.gamma(x_0))**2
+
+        return prefactor * np.exp(exponent)
+
+    def maximum(self, x_0: float) -> float:
+        return 2.0 / (self.gamma(x_0) * np.sqrt(2.0 * np.pi))
+
+
+class Lorentzian(Lineshape):
+    """Lorentzian function on `domain`, centered at `x_0` with broadening `gamma`.
+
+    Parameters
+    ----------
+    domain : Union[List[float], np.ndarray]
+        The domain of the Lorentzian profile.
+    gamma : float
+        Broadening parameter.
+        This is the full width at half maximum (FWHM).
+
+    Notes
+    -----
+    Use this profile to model homogeneous broadening.
+    """
+
+    def lineshape(self, x_0: float) -> np.ndarray:
+        """Lorentzian function on `self.domain`, centered at `x_0` with broadening `self.gamma`.
+
+        Parameters
+        ----------
+        x_0 : float
+            Center of the Lorentzian, i.e. its maximum.
+
+        Returns
+        -------
+        lorentz : np.ndarray
+            The Lorentzian profile.
+        """
+        prefactor = 1.0 / np.pi
+        numerator = self.gamma(x_0) / 2.0
+        denominator = (self.domain - x_0)**2 + numerator**2
+
+        return prefactor * (numerator / denominator)
+
+    def maximum(self, x_0: float) -> float:
+        return 2.0 / (np.pi * self.gamma(x_0))
+
+
+def prefactor_opa() -> float:
+    r"""Prefactor for converting microscopic observable to decadic molar
+    extinction coefficient in one-photon absorption.
+
+    Returns
+    -------
+    prefactor : float
+
+    Notes
+    -----
+    This function implements the calculation of the following prefactor:
+
+    .. math::
+
+        k = \frac{4\pi^{2}N_{\mathrm{A}}}{3\times 1000\times \ln(10) (4 \pi \epsilon_{0}) n \hbar c}
+
+    The prefactor is computed in SI units and then adjusted for the fact that
+    we use atomic units to express microscopic observables: excitation energies
+    and transition dipole moments.
+    The refractive index :math:`n` is, in general, frequency-dependent. We
+    assume it to be constant and equal to 1.
+    """
+
+    N_A = qcel.constants.get("Avogadro constant")
```

Comment:
It's a little safer to `from psi4.driver import constants` and then use as usual. At the moment there's no difference. But that could change if qcel shifts default to codata2018 and psi stays with 2014. At the driver layer, psi can pull an explicit 2014, even when qcel updates default.